File: projects/10/ruby/Ostrich.rb
# Ostrich.rb | Defines the Ostrich class which inherits attributes and methods
#   from the WalkingBird superclass.
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Daniela Contreras
# Date: 03/29/22
####################################################

require './WalkingBird.rb'

class Ostrich < WalkingBird

##############################################
## Call returns sound of Duck                 #
##Return: Quack! a string                    #
##############################################
    def call
        'Snork!'
    end
    
end

File: projects/10/Ostrich.java
/* Ostrich.java provides a Goose subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Daniela Contreras
 * Date: 03/30/22
 ******************************************************/

public class Ostrich extends WalkingBird {

    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
     public Ostrich()
     {
        super();
     }
   
     public Ostrich(String name)
     {     
       super(name);
     } 
   
    /* A Goose's Call
     * Return: a goose-call ("Honk!").
     */
   
     public String call()
     {
       return "Snork!";
     } 
   
   
   }